aria_data_loaders/aria_data_utils/memory_interface/adt_memory_provider.py

```diff
@@ -0,0 +1,645 @@
+
+    def _get_object_context_from_frames(
+        self, found_frames, object_info_dict, debug=False, ground_truth=False
+    ):
+        context = []
+        if ground_truth:
+            for obj_id, frames in found_frames.items():
+                for frame_data in frames:
+                    segmentation = frame_data["segmentation"]
+                    seg_mask = np.zeros_like(segmentation)
+                    seg_mask[segmentation == obj_id] = 1
+                    object_mem = Memory(
+                        name=object_info_dict[obj_id].name,
+                        wearer_location=frame_data[
+                            "pose"
+                        ].transform_scene_device.matrix()[:3, 3],
+                        instance_id=obj_id,
+                        category_name=object_info_dict[obj_id].category,
+                        associated_imgs=np.expand_dims(frame_data["rgb"], axis=0),
+                        associated_3dbbox=np.expand_dims(
+                            frame_data["3dbbox"][obj_id], axis=0
+                        ),
+                        seg_mask=np.expand_dims(
+                            seg_mask,
+                            axis=0,
+                        ),
+                    )
+                    context.append(object_mem)
+                if debug:
+                    breakpoint()
+                    # visualize each segmented image with object in it
+                    object_mem.visualize_objects_with_seg([0])
+        else:
+            for obj_desc, frames in found_frames.items():
+                for frame_data in frames:
+                    # TODO: pass the bbox output from OWL-ViT here
+                    object_mem = Memory(
+                        name=obj_desc,
+                        wearer_location=frame_data[
+                            "pose"
+                        ].transform_scene_device.matrix()[:3, 3],
+                        instance_id=uuid.uuid4(),
+                        category_name=obj_desc,
+                        associated_imgs=np.expand_dims(frame_data["rgb"], axis=0),
+                    )
+                    context.append(object_mem)
+        return context
+
+    def _to_memory(self, object_info) -> Memory:
+        return Memory(
+            name=object_info.name,
+            wearer_location=np.random.rand(3),
+            instance_id=object_info.id,
+            category_name=object_info.category,
+        )
+
+    def _choose_gt_objects(
+        self, num_instances: int = 1
+    ) -> Tuple[List[int], List[dict]]:
+        (
+            object_info_dict,
+            object_instance_ids,
+        ) = self.data_loader.get_all_annotated_objects()
+        chosen_object_ids = np.random.choice(object_instance_ids, num_instances)
+        chosen_objects = [object_info_dict[obj_id] for obj_id in chosen_object_ids]
+        return chosen_objects, [object_info_dict[oid] for oid in chosen_object_ids]
+
+    def create_object_retrieval_benchmark(
```

Comment:
Saw this function in ADT data loader too, potential code duplication?

Reply:
Thanks for pointing out. This is the right place, deleted from ADT data loader. That
class should be independent of application.